Source/CarthageKit/Project.swift

```diff
@@ -769,7 +762,7 @@ public final class Project { // swiftlint:disable:this type_body_length
 			.zip(with: submodulesSignal)
 			.flatMap(.merge) { dependencies, submodulesByPath -> SignalProducer<(), CarthageError> in
 				return SignalProducer<(Dependency, PinnedVersion), CarthageError>(dependencies)
-					.map { (dependency, version) -> SignalProducer<(), CarthageError> in
+					.flatMap(.concurrent(limit: 4)) { (dependency, version) -> SignalProducer<(), CarthageError> in
```

Comment:
@mdiep [suggested to limit the concurrency here](https://github.com/Carthage/Carthage/pull/2082#discussion_r129959730).

Unless, maybe I misinterpreted your comment — are you suggesting to `limit` with `min(ProcessInfo.processInfo.activeProcessorCount, 4)`?